assets/js/app/LocusZoom.js

```diff
@@ -319,42 +319,54 @@ LocusZoom.parseFields = function (data, html) {
     if (typeof html != "string"){
         throw ("LocusZoom.parseFields invalid arguments: html is not a string");
     }
-    // Handle conditional blocks {{#if field}}...{{/if}} from back to front to handle nesting correctly
-    var if_regex = /\{\{#if ([0-9A-Za-z_:|]+)\}\}/g;
-    var fi_regex = /^(.*?)\{\{\/if\}\}(.*)/;
-    while(1) {
-        var m, match = false;
-        while(m = if_regex.exec(html)) { match = m; }
-        if (!match) break;
-        var before = html.slice(0, match.index);
-        var condition = match[1];
-        var x = fi_regex.exec(html.slice(match.index + match[0].length));
-        if (!x) { throw("failed to find {{/if}} after {{#if " + condition + "}}"); }
-        var then = x[1];
-        var rest = x[2];
-        if ((new LocusZoom.Data.Field(condition)).resolve(data)) {
-            html = before + then + rest;
-        } else {
-            html = before + rest;
-        }
-    }
-    // Match all things that look like fields in the HTML
-    var matches = html.match(/{{[A-Za-z0-9_:|]+}}/g);
-    if (matches){
-        // Remove duplicates
-        matches.reduce(function(a,b){if(a.indexOf(b)<0)a.push(b);return a;},[]);
-        // Replace matches with resolved values from the data object
-        matches.forEach(function(match){
-            var field = new LocusZoom.Data.Field(match.substring(2, match.length-2));
-            var value = field.resolve(data);
-            if (["string","number","boolean"].indexOf(typeof value) != -1){
-                html = html.replace(match, value);
-            } else if (value == null){
-                html = html.replace(match, "");
-            }
-        });
+    // `tokens` is like [token,...]
+    // `token` is like {text: '...'} or {variable: 'foo|bar'} or {condition: 'foo|bar'} or {close: 'if'}
+    var tokens = [];
+    var regex = /\{\{(?:(#if )?([A-Za-z0-9_:\|]+)|(\/if))\}\}/;
+    while (html.length > 0){
+        var m = regex.exec(html);
+        if (!m) { tokens.push({text: html}); html = ''; }
+        else if (m.index != 0) { tokens.push({text: html.slice(0, m.index)}); html = html.slice(m.index); }
+        else if (m[1] == "#if ") { tokens.push({condition: m[2]}); html = html.slice(m[0].length); }
+        else if (m[2]) { tokens.push({variable: m[2]}); html = html.slice(m[0].length); }
+        else if (m[3] == "/if") { tokens.push({close: 'if'}); html = html.slice(m[0].length); }
+        else { throw [m, html] }
```

Comment:
I thought it was probably okay to have `throw` in `parseFields` in #91 but I actually think it'd be better to wrap this while loop in a try/catch and handle any thrown exceptions within the method. The reason here is that it's not the core library that's failing, we're just trying to parse some badly marked up HTML.

At the very least we should catch any such exceptions and pipe them gracefully to the console, then try our best to parse and return the HTML we've been given with any bad syntax untouched. This way `parseFields` can be trusted to always return a string, so in the case of tool tips, a tool tip will always render (but someone trying to make LZ do something clever will see unparsed stuff in their tool tips with detail in the console instead of an uncaught exception).